site/app/templates/course/CourseMaterials.twig

```diff
@@ -56,96 +58,118 @@
 
-        }
-        else {
-            // clean up all cookies which are stated with "cm_";
-            $('[id^=div_viewer_]').each(function() {
-            var number = this.id.replace('div_viewer_', '').trim();
-            var keyValuePairs = document.cookie.split(';');
-
-            for(var i = 0; i < keyValuePairs.length; i++) {
-                var name = keyValuePairs[i].substring(0, keyValuePairs[i].indexOf('=')).trim();
-                if (name === 'cm_'+number) {
-                    document.cookie = 'cm_' + number + '=';
+        } else { // clean up all cookies which are stated with "cm_";
+            $('[id^=div_viewer_]').each(function () {
+                var number = this.id.replace('div_viewer_', '').trim();
+                var keyValuePairs = document.cookie.split(';');
+
+                for (var i = 0; i < keyValuePairs.length; i++) {
+                    var name = keyValuePairs[i].substring(0, keyValuePairs[i].indexOf('=')).trim();
+                    if (name === 'cm_' + number) {
+                        document.cookie = 'cm_' + number + '=';
+                    }
                 }
-            }
-        });
+            });
         }
     }
 
-    function shareToOther(id, path) {
-        // pass filename to server to record the permission of the file
+    function shareToOther(id, path) { // pass filename to server to record the permission of the file
         var idName = "#share_checkbox_" + id;
-        if($(idName).is(':checked')) {
+        if ($(idName).is(':checked')) {
             changePermission(path, '1');
-        }
-        else {
+        } else {
             changePermission(path, '0');
         }
     }
 
-    function setNewDateTime(me, path) {
-        // pass filename to server to record the new date and time of the file to be released
+    function setNewDateTime(me, path) { // pass filename to server to record the new date and time of the file to be released
         var newDateTime = $(me).val();
-        //alert(newDateTime);
+        // alert(newDateTime);
         changeNewDateTime(path, newDateTime);
     }
 
-    $(document).ready(function() {
+    $(document).ready(function () {
         flatpickr(".date_picker", {
+            plugins: [ShortcutButtonsPlugin(
+                    {
+                        button: [
+                            {
+                                label: "Now"
+                            }, 
+                            {
+                                label: "End of time"
+                            }
+                        ],
+                        label: "or",
+                        onClick: (index, fp) => {
+                            let date;
+                            switch (index) {
+                                case 0: date = new Date();
+                                    break;
+                                case 1: date = new Date("9999-01-01 00:00:00");
+                                    break;
```

Comment:
Put the line of code on the next line after the case line (indented).

Reply:
Done